third_party/3/jwt/algorithms.pyi

```diff
@@ -1,3 +1,43 @@
-from typing import Any
+import sys
+from hashlib import _Hash
+from typing import Any, Set, Dict, Optional, ClassVar, Union, Generic, TypeVar
 
-class Algorithm(Any): ...  # type: ignore
+requires_cryptography = Set[str]
+
+def get_default_algorithms() -> Dict[str, Algorithm]: ...
+
+_K = TypeVar("_K")
+
+class Algorithm(Generic[_K]):
+    def prepare_key(self, key: _K) -> _K: ...
+    def sign(self, msg: bytes, key: _K) -> bytes: ...
+    def verify(self, msg: bytes, key: _K, sig: bytes) -> bool: ...
+    @staticmethod
+    def to_jwk(key_obj: _K) -> str: ...
+    @staticmethod
+    def from_jwk(jwk: str) -> _K: ...
```

Comment:
I don't think this works correctly:
```
$ cat bin/genstat.py 
from typing import *

T = TypeVar('T')

class A(Generic[T]):
    @staticmethod
    def foo() -> T:
        pass

class B(A[int]):
    pass

reveal_type(B.foo())
$ python3.7 -m mypy bin/genstat.py 
bin/genstat.py:13: error: Revealed type is 'T`1'
```
This is python/mypy#1337.

There doesn't appear to be a good workaround, so we may have to use `Any` for now.